### PR TITLE
Temperature sensor

### PR DESCRIPTION
**Pull request by emilybthorpe**

### CODE REVIEW

**Comment by ntlhui** [comment]
@emilybthorpe please resolve conflicts

File: src/cli/menuItems/debugCommands.cpp [hunk 1]
```diff
@@ -62,4 +67,38 @@ void CLI_testPrintf(void)
...
+			}
+		}
+        
+        temp = pSystemDesc->pTempSensor->getTemp();
+        SF_OSAL_printf("Temperature Reading: %f" __NL__, temp);
+    }
+
+    SF_OSAL_printf(__NL__);
+    pSystemDesc->pTempSensor->stop();
 }
```
Comment (ntlhui) [inline src/cli/menuItems/debugCommands.cpp:104]: Missing trailing newline

**Review by ntlhui — CHANGES_REQUESTED** [review]

File: src/cli/cliDebug.cpp [hunk 2]
```diff
@@ -43,22 +48,23 @@ void CLI_doDebugMode(void)
...
         getline(userInput, SF_CLI_MAX_CMD_LEN);
+        
         if (atoi(userInput) == 0) // Exiting debug mode
         {
             break;
         }
 
-        SF_OSAL_printf("\r\n");
+        SF_OSAL_printf("\r" __NL__);
+
```
[the remark this answers is not in the document]
Comment (emilybthorpe) [reply src/cli/cliDebug.cpp:58]: Fixed

File: src/cli/flog.hpp [hunk 3]
```diff
@@ -34,9 +34,11 @@ typedef enum FLOG_CODE_
     FLOG_UPL_BATT_LOW     =0x0602,
     FLOG_UPL_FOLDER_COUNT =0x0603,
     FLOG_UPL_CONNECT_FAIL =0x0604,
+    FLOG_RESET_REASON     =0x0605,
     FLOG_GPS_INIT_FAIL    =0x0605,
     FLOG_GPS_START_FAIL   =0x0606,
-    FLOG_CHARGER_REMOVED  =0x0701
+    FLOG_TEMP_FAIL        =0x0701,
+    FLOG_CHARGER_REMOVED  =0x0702
```
[the remark this answers is not in the document]
Comment (emilybthorpe) [reply src/cli/flog.hpp:41]: Sounds good

File: src/cli/menuItems/debugCommands.cpp [hunk 4]
```diff
@@ -62,4 +67,38 @@ void CLI_testPrintf(void)
...
+			ch = getch();
+			if('q' == ch) 
+			{
+				break;
+			}
+		}
+        
+        temp = pSystemDesc->pTempSensor->getTemp();
+        SF_OSAL_printf("Temperature Reading: %f" __NL__, temp);
+    }
```
[the remark this answers is not in the document]
Comment (emilybthorpe) [reply src/cli/menuItems/debugCommands.cpp:100]: Fixed

File: src/cli/menuItems/debugCommands.cpp (hunk 1 above)
Comment (emilybthorpe) [reply src/cli/menuItems/debugCommands.cpp:104]: Fixed

File: src/system.cpp [hunk 5]
```diff
@@ -64,6 +79,15 @@ static int SYS_initTasks(void)
...
+
+
+static int SYS_initTempSensor(void)
+{
+    Wire.begin();
+    systemDesc.pTempSensor = &tempSensor;
+
     ledTimer.start();
 
     return 1;
```
[the remark this answers is not in the document]
Comment (emilybthorpe) [reply src/system.cpp:93]: Removed

File: src/temperature/max31725.h
[the remark this answers is not in the document]
Comment (emilybthorpe) [reply src/temperature/max31725.h:1]: I'm not sure where the source or what to mark the liscence as, as well as there is no version number listed. What would you suggest having?

[the remark this answers is not in the document]
Comment (ntlhui) [reply src/temperature/max31725.h:1]: https://os.mbed.com/teams/MaximIntegrated/code/MAX31725_Accurate_Temperature_Sensor//file/b4fdbbe79036/max31725.h/

File: README.md [hunk 6]
```diff
@@ -10,6 +10,9 @@ gps-quectel     | 1.0.0   | Apache 2.0 | https://github.com/particle-iot/gps-que
 gps-nmea-parser | 1.2.0   | Apache 2.0 | https://github.com/particle-iot/gps-nmea-parser/ | lib/gps-nmea-parser
 tracker edge    | v18     | Apache 2.0 | https://github.com/particle-iot/tracker-edge     | lib/tracker-edge
 SparkFun ICM    | 1.2.0   | MIT     | https://github.com/UCSD-E4E/SparkFun_ICM-20948_ParticleLibrary | lib/SparkFun_ICM-20948_ParticleLibrary
+Max 31725       | 2019    | Copyright Max 2019 | https://os.mbed.com/teams/MaximIntegrated/code/MAX31725_Accurate_Temperature_Sensor//file/b4fdbbe79036/max31725.h/ | temperature/
```
Comment (ntlhui) [inline README.md:13]: Per that website, all code is licensed under Apache.

**Review by ntlhui — APPROVED** [review]

**Review by ntlhui — APPROVED** [review]

**Pull request merged**